### PR TITLE
[SPARK-33649][SQL][DOC] Improve the doc of spark.sql.ansi.enabled

### PR DESCRIPTION
**Pull request by gengliangwang**
### What changes were proposed in this pull request?

Improve the documentation of SQL configuration `spark.sql.ansi.enabled`

### Why are the changes needed?

As there are more and more new features under the SQL configuration `spark.sql.ansi.enabled`, we should make it more clear about:
1. what exactly it is
2. where can users find all the features of the ANSI mode
3. whether all the features are exactly from the SQL standard

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

It's just doc change.

### CODE REVIEW

**Comment by gengliangwang** [comment]
cc @bart-samwel as well

File: docs/sql-ref-ansi-compliance.md [hunk 1]
```diff
@@ -21,7 +21,8 @@ license: |
 
 Since Spark 3.0, Spark SQL introduces two experimental options to comply with the SQL standard: `spark.sql.ansi.enabled` and `spark.sql.storeAssignmentPolicy` (See a table below for details).
 
-When `spark.sql.ansi.enabled` is set to `true`, Spark SQL follows the standard in basic behaviours (e.g., arithmetic operations, type conversion, SQL functions and SQL parsing).
+When `spark.sql.ansi.enabled` is set to `true`, Spark SQL Spark SQL uses an ANSI compliant dialect instead of being Hive compliant. For example, Spark will throw an exception at runtime instead of returning null results if the inputs to a SQL operator/function are invalid. Some ANSI dialect features may not be from the ANSI SQL standard directly, but their behaviors align with ANSI SQL's style.
```
Comment (cloud-fan) [inline docs/sql-ref-ansi-compliance.md:24]: `Spark SQL Spark SQL` -> `Spark SQL`

Comment (cloud-fan) [reply docs/sql-ref-ansi-compliance.md:24]: `may not be` -> `may not`

**Review by cloud-fan — APPROVED** [review]
Thanks for improving it!

File: docs/sql-ref-ansi-compliance.md (hunk 1 above)
Comment (gengliangwang) [reply docs/sql-ref-ansi-compliance.md:24]: > may not be -> may not


We need to have a verb in the sentence.

**Review by HyukjinKwon — APPROVED** [review]

**Comment by gengliangwang** [comment]
The test failure is not relevant. Merging this one to the Master branch since it is just doc change.

**Pull request closed without merging**